test/NuGet.Core.FuncTests/Dotnet.Integration.Test/ProjectFileBuilder.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Dotnet.Integration.Test;
+
+namespace NuGet.Test.Utility
+{
+    /// <summary>
+    /// Represents a builder for dotnet core projects for integration tests.
+    /// Relies on <c>MsbuildIntegrationTestFixture</c> to programmatically create the project
+    /// </summary>
+    /// <seealso cref="MsbuildIntegrationTestFixture"/>
+    public class ProjectFileBuilder
+    {
+        public string PackageIcon { get; private set; }
+        public string PackageIconUrl { get; private set; }
+        public string ProjectName { get; private set; }
+        public List<Tuple<string, string, string>> ItemGroupEntries { get; private set; }
```

Comment:
1. `ItemGroupEntries` doesn't appear to be used outside of this class, so why it is public? Same question about the other properties. Only `ProjectName` and `ProjectFilepath` appear to be used. (you could have `Build` return this, rather than making it class properties, so the builder can be GCed after `Build` is called)
2. In my opinion, `Tuple<>` should never be used on anything public because `.Item1`, `.Item2` are meaningless. Others reading the code can't possibly get enough information to understand what it means.
3. If you really don't want to create a struct or class, at least used a [named tuple](https://docs.microsoft.com/en-us/dotnet/csharp/tuples)

Reply:
2/3. I will add the Model class for these entries. It will help to simplify some code.